packages/labs/virtualizer/src/virtualize.ts

```diff
@@ -74,7 +74,11 @@ class VirtualizeDirective<T = unknown> extends AsyncDirective {
     const itemsToRender: Array<T> = [];
 
     if (this._first >= 0 && this._last >= this._first) {
-      for (let i = this._first; i <= this._last; i++) {
+      for (
+        let i = this._first;
+        i <= this._last && i < this._items.length;
+        i++
+      ) {
         itemsToRender.push(this._items[i]);
```

Comment:
I sometimes got `undefined` elements in `keyFunction` and `renderItem`. I think when item set change with less items, _last is recomputed after a render event, or something like that. This avoid `undefined` values. Maybe it would require a fix before being here.

Reply:
I think this is this issue https://github.com/lit/lit/issues/4833